### ISSUE TITLE
Checkov not using custom policy

### ISSUE THREAD

**Issue by Z3r0-gr33n** [bridgecrewio/checkov#4930]
**Describe the issue**
I am working on a POC and began testing a policy that already exist to ensure I get an understanding of the language. When I run the following command it does not trigger my custom policy.

**Command to run the check**
Returns empty finding:
`checkov -d . --external-checks-dir ./policy/build-policy.yaml -c CUSTOM_AZ-1  `

Returns finding but only built in findings not my custom policy:
`checkov -d . --external-checks-dir ./policy/build-policy.yaml`

My folder structure is as follows:
```
/test
   main.tf
   policy/build_policy.yaml
```

**Examples**
My test policy is as follows

**Checkov Policy**
```
metadata:
  name: "CUSTOM_AZ-1"
  id: "CUSTOM_AZ-1"
  category: "compute"
definition:
  and:
    - attribute: resource_type
      cond_type: filter
      operator: within
      value:
        - azurerm_cosmosdb_account
    - attribute: public_network_access_enabled
      cond_type: attribute
      operator: exists
      resource_types:
        - azurerm_cosmosdb_account
    - attribute: public_network_access_enabled
      cond_type: attribute
      resource_types:
        - azurerm_cosmosdb_account
     operator: is_true
```
For my terraform code I am just using an azurerm_cosmosdb_account from terraforms site just to test the policy and is as follows:

**Terraform Code**
```
resource "azurerm_resource_group" "example" {
  name     = "example-resource-group"
  location = "West Europe"
}

resource "random_integer" "ri" {
  min = 10000
  max = 99999
}

resource "azurerm_cosmosdb_account" "db" {
  name                = "tfex-cosmos-db-${random_integer.ri.result}"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  offer_type          = "Standard"
  kind                = "MongoDB"

  public_network_access_enabled = true

  enable_automatic_failover = true

  capabilities {
    name = "EnableAggregationPipeline"
  }

  capabilities {
    name = "mongoEnableDocLevelTTL"
  }

  capabilities {
    name = "MongoDBv3.4"
  }

  capabilities {
    name = "EnableMongo"
  }

  consistency_policy {
    consistency_level       = "BoundedStaleness"
    max_interval_in_seconds = 300
    max_staleness_prefix    = 100000
  }

  geo_location {
    location          = "eastus"
    failover_priority = 1
  }

  geo_location {
    location          = "westus"
    failover_priority = 0
  }
}
```

**Version (please complete the following information):**
 - Checkov Version 2.3.183

**Additional context**

**Comment by gruebel** [comment]
hey @Z3r0-gr33n thanks for reaching out.

You need to reference the folder not the file
```shell
checkov -d . --external-checks-dir ./policy -c CUSTOM_AZ-1
```
I also recommend you to not put the scanned files in the same folder as the custom policy, because `checkov` will try to scan your custom policy to find out, if it is an IaC file.

**Comment by Z3r0-gr33n** [comment]
That was it I appreciate the response!

**Comment by ritesh1735** [comment]
Hi I am running below command 
`checkov -f ec2.tf --external-checks-dir custom_policy/`
and my policy as below
`metadata:
  name: "instance_sizes limiter"
  id: "CKV2_AWS_999"
  guidelines: "It defines what type of instance should be used"
  category: "GENERAL_SECURITY" 
  severity: "INFO" 
scope:
    provider: "aws" 
definition:
  and:
    - cond_type: "attribute"
      resource_types:
        - "aws_instance"
      attribute: "instance_type"
      operator: "equals"
      value: "t3.micro"`
      
      But Doesn't show result for the above custom policy at the same time If I am running below policy it is working fine 
      `---      
metadata:
  name: "Check EC2 has special Tags"
  id: "CKV2_AWS_999"
  category: "GENERAL_SECURITY"
  guideline: "Please add required tags to EC2"
scope:
  provider: aws

definition:
  and:
    - cond_type: attribute
      resource_types:
        - "aws_instance"
      attribute: "tags.project"
      operator: "exists"
    - cond_type: attribute
      resource_types:
        - "aws_instance"
      attribute: "tags.owner"
      operator: "exists"`
      Do anyone havingh idea about it??
